src/__tests__/BoardSizeForm.test.tsx

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+import { shallow, mount } from 'enzyme';
+import BoardSizeForm from '../components/BoardSizeForm';
+
+describe('BoardSizeForm component', () => {
+  it('renders a form', () => {
+    const wrapper = shallow(
+      <BoardSizeForm
+        handleChange={jest.fn()}
+        handleSubmit={jest.fn()}
+        hasError={false}
+      />
+    );
+    expect(wrapper.exists('#size-form')).toBeTruthy();
+  });
+
+  it('renders an error if given one', () => {
+    const wrapper = mount(
+      <BoardSizeForm
+        handleChange={jest.fn()}
+        handleSubmit={jest.fn()}
+        hasError={true}
+      />
+    );
+    expect(
+      wrapper.contains(
+        <p>Error: Invalid input. Please input a number from 3 - 10</p>
+      )
+    ).toBeTruthy();
+  });
+
+  it('calls its handleChange method', () => {
+    const mockFn = jest.fn();
+    const wrapper = shallow(
+      <BoardSizeForm
+        handleChange={mockFn}
+        handleSubmit={jest.fn()}
+        hasError={false}
+      />
+    );
+    expect(mockFn.mock.calls).toHaveLength(0);
+    wrapper
+      .find('#size-input')
+      .simulate('change', { target: { value: '123' } });
```

Comment:
Did you figure out how to get the event argument passed to the handler? If not, it's dead code, and I'd remove it.